specification/configuration.md

```diff
@@ -130,6 +136,34 @@ beyond the OpenTelemetry specification exist.
   - Distribution MUST default to `otlp`
   - Distribution MUST offer `jaeger-thrift-splunk` that defaults to `http://127.0.0.1:9080/v1/trace`
 
+#### Real User Monitoring Libraries
+
+Real User Monitoring (RUM) instrumentation libraries cannot use environment variables for configuration.
+Instead, they MUST expose a `SplunkRum` object/class/namespace (depending on the language used) that allows
+setting the properties listed below (in a language-specific way: Java may use builders, Swift can use named
+parameters with default values, JavaScript can use objects, etc.).
+
+RUM instrumentation libraries MUST support the following configuration properties:
+
+| Property (default value)               | Description                                                                                                                                                                                                 |
+| -------------------------------------- | -----------                                                                                                                                                                                                 |
+| `realm` ()                             | Splunk realm, e.g. `us0`, `us1`. If set, value of `beaconEndpoint` will be automatically computed based on this. [1] [2] [3]                                                                                |
+| `beaconEndpoint` ()                    | RUM beacon URL, e.g. `https://rum-ingest.<realm>.signalfx.com/v1/rum`. If both `realm` and `beaconEndpoint` are set, `beaconEndpoint` takes precedence. [1] [2] [3]                                         |
+| `rumAccessToken` ()                    | RUM authentication token. [1]                                                                                                                                                                               |
+| `serviceName` ()                       | Instrumented application name. [1]                                                                                                                                                                          |
```

Comment:
I don't think this is a good configuration value. I think it will be very confusing for mobile developers, since the app isn't a "service", and they will probably think this is referring to some backend service that the app might be using. I think we should use `applicationName` or `appName` or `app` instead.

Reply:
(sorry for the late comment..I was going through to implement what was in this PR and ran into this, which had changed since I last looked in detail)

Reply:
I agree with you and I think that `applicationName` is a much better name - `serviceName` sounds really confusing in the context of RUM.
@flands I think you were the only person that preferred `serviceName`. Can we change that to a less confusing `applicationName`?

Reply:
I'm guessing the ask for this to be `serviceName` was from a desire to be consistent with other parts of the specification. However, given this part of the specification in only for RUM libraries I think it is prudent to ask these kinds of questions and make sure we provide them with names that are meaningful and allow the least amount of cognitive load when they start using our services.

@flands any opposition?

Reply:
Did we really decide to stick with `serviceName`? I think this is a very bad idea, as stated above.

Reply:
ah never mind. Looks like I was looking at at old version. :disappear: